Assets/MRTK/SDK/Experimental/Drawstring/Scripts/TwoHandleManipulator.cs

```diff
@@ -0,0 +1,937 @@
+        {
+            get => onHoverExited;
+            set => onHoverExited = value;
+        }
+        #endregion
+
+        #region Private Properties
+
+        private ManipulationMoveLogic moveLogic;
+        private TwoHandScaleLogic scaleLogic;
+        private TwoHandRotateLogic rotateLogic;
+        private ElasticHandleLogic elasticLogic;
+
+        private Vector3 leftLineOffset, rightLineOffset;
+
+        private float lastUpdateTime = 0.0f;
+
+
+        /// <summary>
+        /// Holds the pointer and the initial intersection point of the pointer ray 
+        /// with the object on pointer down in pointer space
+        /// </summary>
+        private struct PointerData
+        {
+            public IMixedRealityPointer pointer;
+            public HandleSide side;
+            private Vector3 initialGrabPointInPointer;
+
+            public PointerData(IMixedRealityPointer pointer, Vector3 worldGrabPoint, HandleSide side) : this()
+            {
+                this.pointer = pointer;
+                this.side = side;
+                this.initialGrabPointInPointer = Quaternion.Inverse(pointer.Rotation) * (worldGrabPoint - pointer.Position);
+            }
+
+            public bool IsNearPointer => pointer is IMixedRealityNearPointer;
+
+            /// Returns the grab point on the manipulated object in world space
+            public Vector3 GrabPoint => (pointer.Rotation * initialGrabPointInPointer) + pointer.Position;
+        }
+
+        private Dictionary<uint, PointerData> pointerIdToPointerMap = new Dictionary<uint, PointerData>();
+        private Quaternion objectToGripRotation;
+        private bool isNearManipulation;
+        private bool isManipulationStarted;
+
+        private Rigidbody rigidBody;
+        private bool wasKinematic = false;
+
+        private ConstraintManager constraints;
+
+        private bool IsOneHandedManipulationEnabled => manipulationType.HasFlag(ManipulationHandFlags.OneHanded) && pointerIdToPointerMap.Count == 1;
+        private bool IsTwoHandedManipulationEnabled => manipulationType.HasFlag(ManipulationHandFlags.TwoHanded) && pointerIdToPointerMap.Count > 1;
+
+        #endregion
+
+        #region MonoBehaviour Functions
+
+        private void Awake()
+        {
+            moveLogic = new ManipulationMoveLogic();
+            rotateLogic = new TwoHandRotateLogic();
+            elasticLogic = new ElasticHandleLogic();
+        }
+        private void Start()
+        {
+            rigidBody = HostTransform.GetComponent<Rigidbody>();
+            constraints = new ConstraintManager(gameObject);
+
+            leftLineOffset = lineDataProvider.transform.localPosition - leftHandle.localPosition;
+            rightLineOffset = (lineDataProvider.transform.localPosition + lineDataProvider.EndPoint.Position) - rightHandle.localPosition;
+            
+        }
+
+        private void Update()
+        {
+            if(pointerIdToPointerMap.Count == 0)
```

Comment:
```suggestion
            if (pointerIdToPointerMap.Count == 0)
```

Reply:
In general try using dotnet format on this file:
https://github.com/dotnet/format